Indicators/TimeSeriesForecast.cs

```diff
@@ -0,0 +1,89 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+namespace QuantConnect.Indicators
+{
+/// <summary>
+/// Represents an indicator capable of predicting new values given previous data from a window.
+/// Source: https://tulipindicators.org/tsf
+/// </summary>
+public class TimeSeriesForecast : WindowIndicator<IndicatorDataPoint>, IIndicatorWarmUpPeriodProvider
+{
+    /// <summary>
+    /// Gets a flag indicating when this indicator is ready and fully initialized
+    /// </summary>
+    public override bool IsReady => Samples >= Period;
+
+    /// <summary>
+    /// Required period, in data points, for the indicator to be ready and fully initialized
+    /// </summary>
+    public int WarmUpPeriod => Period;
```

Comment:
I don't think u need to override `WarmUpPeriod/IsReady` because their value/behavior is the same as in the base class? can just rely on the inheritance

Reply:
Thanks! Yes you are correct.